helix-term/src/commands/typed.rs

```diff
@@ -13,6 +13,8 @@ pub struct TypableCommand {
     // params, flags, helper, completer
     pub fun: fn(&mut compositor::Context, &[Cow<str>], PromptEvent) -> anyhow::Result<()>,
     pub completer: Option<Completer>,
+    /// Use `prompt_events` to declare which events this command will respond to.
+    pub prompt_events: u8,
```

Comment:
Is this really worth it? It adds extra complexity. I'd just always call the prompt

Reply:
And then return early inside the function. That way it behaves exactly like regex_prompt and other command prompts.

Reply:
I had that initially, but swapped to this approach to avoid touching the existing functions: https://github.com/helix-editor/helix/pull/1798#discussion_r829672711

I don't think it's going to be a clean revert at this point unfortunately
https://github.com/helix-editor/helix/pull/1798/commits/8dfd8403ef0270d656647e52ab442dd95272d142

I don't have string opinions on either approach. If you confirm that you'd prefer what was done originally I'll change it back, but I'll only do it one more time or else I might get tendonitis 😳

Reply:
Switched it back, should be good to go!